### PR TITLE
Separate round change reception from RoundChangeCertificate

### PR DESCRIPTION
**Pull request by rain-on**
IBFT2.1 requires that message content be separated, thus concepts
at the message level must not leak into the business logic - eg
RoundChangeCertificate should not be created by the
RoundChangeManager - rather an intermediate type is to be inserted.

### CODE REVIEW

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/RoundChangeArtefacts.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  public static RoundChangeArtefacts create(final Collection<RoundChange> roundChanges) {
+
+    final Collection<SignedData<RoundChangePayload>> payloads =
+        roundChanges
+            .stream()
+            .map(roundChange -> roundChange.getSignedPayload())
+            .collect(Collectors.toList());
+
+    final Optional<PreparedCertificate> latestdPreparedCertificate =
```
Comment (jframe) [inline consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/RoundChangeArtefacts.java]: nit: typo in latestdPreparedCertificate

Comment (rain-on) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/RoundChangeArtefacts.java]: done.

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/IbftBlockInterface.java [hunk 2]
```diff
@@ -49,4 +52,33 @@ public Address getProposerOfBlock(final BlockHeader header) {
...
     return ibftExtraData.getValidators();
   }
+
+  public int roundOfBlock(final BlockHeader header) {
+    final IbftExtraData ibftExtraData = IbftExtraData.decode(header.getExtraData());
+    return ibftExtraData.getRound();
+  }
+
+  public static Block replaceRoundInBlock(
+      final Block block, final int round, final BlockHashFunction blockHashFunction) {
```
[the remark this answers is not in the document]
Comment (rain-on) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/IbftBlockInterface.java]: done.

File: consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/IbftBlockInterface.java [hunk 3]
```diff
@@ -49,4 +52,33 @@ public Address getProposerOfBlock(final BlockHeader header) {
     final IbftExtraData ibftExtraData = IbftExtraData.decode(header.getExtraData());
     return ibftExtraData.getValidators();
   }
+
+  public int roundOfBlock(final BlockHeader header) {
```
[the remark this answers is not in the document]
Comment (rain-on) [reply consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/IbftBlockInterface.java]: removed.

**Pull request merged**